quill-core/src/main/scala/io/getquill/Query.scala

```diff
@@ -47,6 +47,7 @@ sealed trait EntityQuery[T]
   extends Query[T] {
 
   def apply(alias: String, propertyAlias: (T => (Any, String))*): EntityQuery[T]
+  def schema(f: Schema[T] => Any): EntityQuery[T]
```

Comment:
I haven't tested `apply` with Cql but if it's already working we should also add support for this in the CqlIdiom. The only thing that is not support by Cassandra is the generation of keys, but that's not a big deal.